### ISSUE TITLE
Stuck Unlocking Wallet 

### ISSUE THREAD

**Issue by corbimos** [lightning-power-users/node-launcher#345]
* **I'm submitting a ...**
  - [ ] support request

* **What is the current behavior?**
lnd is stuck unlocking the wallet. It was stuck in a loop, so i deleted the keychain hoping i could enter the information manually. I have the passwords and seed backed up. 

The output is stuck at:

> 2019-05-16 19:20:07.999 [INF] LTND: Version: 0.6.0-beta commit=v0.6-beta, build=production, logging=default
2019-05-16 19:20:08.000 [INF] LTND: Active chain: Bitcoin (network=mainnet)
2019-05-16 19:20:08.001 [INF] CHDB: Checking for schema update: latest_version=8, db_version=8
2019-05-16 19:20:08.011 [INF] RPCS: password RPC server listening on 127.0.0.1:10009
2019-05-16 19:20:08.011 [INF] RPCS: password gRPC proxy started at 127.0.0.1:8080
2019-05-16 19:20:08.012 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create a wallet, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.

Tried to use console to unlock it but that won't allow me to enter the password.

> > unlock
Input wallet password:
[lncli] inappropriate ioctl for device

* **Please tell us about your environment:**

  - Operating System: macOSX 10.14

* **Other information** 

Just to test, I renamed my original lnd folder to lnd.old and restarted. Everything worked fine after it synced, but obviously it just created a new wallet, not the one i need. It sticks at unlocking when i use the original lnd folder.

**Comment by corbimos** [comment]
Ok, i got it to unlock by going through terminal, not console to type the password. 

It says it unlocks successfully but then shuts down. 

`2019-05-16 20:00:42.890 [INF] LTND: Version: 0.6.0-beta commit=v0.6-beta, build=production, logging=default
2019-05-16 20:00:42.890 [INF] LTND: Active chain: Bitcoin (network=mainnet)
2019-05-16 20:00:42.891 [INF] CHDB: Checking for schema update: latest_version=8, db_version=8
2019-05-16 20:00:42.901 [INF] RPCS: password RPC server listening on 127.0.0.1:10009
2019-05-16 20:00:42.901 [INF] RPCS: password gRPC proxy started at 127.0.0.1:8080
2019-05-16 20:00:42.902 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create a wallet, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.
2019-05-16 20:02:20.895 [INF] LNWL: Opened wallet
2019-05-16 20:02:21.022 [INF] LTND: Primary chain is set to: bitcoin
2019-05-16 20:02:21.056 [INF] LTND: Initializing bitcoind backed fee estimator
2019-05-16 20:02:21.056 [INF] LNWL: Started listening for bitcoind block notifications via ZMQ on tcp://127.0.0.1:18500
2019-05-16 20:02:21.056 [INF] LNWL: Started listening for bitcoind transaction notifications via ZMQ on tcp://127.0.0.1:18501
2019-05-16 20:02:21.877 [INF] LNWL: The wallet has been unlocked without a time limit
2019-05-16 20:02:21.881 [INF] LTND: LightningWallet opened
2019-05-16 20:02:21.967 [INF] HSWC: Restoring in-memory circuit state from disk
2019-05-16 20:02:21.982 [INF] LNWL: Started rescan from block 00000000000000000013e3ddf285abd052fca080c09e79bef1d637405d13ec37 (height 564672) for 24 addresses
2019-05-16 20:02:21.983 [INF] LNWL: Starting rescan from block 00000000000000000013e3ddf285abd052fca080c09e79bef1d637405d13ec37
2019-05-16 20:02:21.994 [INF] HSWC: Payment circuits loaded: num_pending=0, num_open=0
2019-05-16 20:02:21.999 [INF] HSWC: Trimming open circuits for chan_id=562360:3077:0, start_htlc_id=0
2019-05-16 20:02:21.999 [INF] HSWC: Trimming open circuits for chan_id=562236:1668:0, start_htlc_id=26
2019-05-16 20:02:21.999 [INF] LNWL: Rescan finished at 564672 (00000000000000000013e3ddf285abd052fca080c09e79bef1d637405d13ec37)
2019-05-16 20:02:21.999 [ERR] LNWL: Unable to complete chain rescan: -1: Block not available (pruned data)
2019-05-16 20:02:21.999 [INF] LNWL: Catching up block hashes to height 564672, this might take a while
2019-05-16 20:02:22.010 [INF] LTND: Received terminated
2019-05-16 20:02:22.012 [INF] LTND: Shutting down...
2019-05-16 20:02:22.012 [INF] LTND: Gracefully shutting down.
2019-05-16 20:02:22.051 [INF] LNWL: Done catching up block hashes
2019-05-16 20:02:22.051 [INF] LNWL: Finished rescan for 24 addresses (synced to block 00000000000000000013e3ddf285abd052fca080c09e79bef1d637405d13ec37, height 564672)
2019-05-16 20:02:22.143 [INF] LTND: Channel backup proxy channel notifier starting
2019-05-16 20:02:22.143 [INF] ATPL: Instantiating autopilot with cfg: (*main.autoPilotConfig)(0xc0000408c0)({
Active: (bool) false,
Heuristic: (map[string]float64) (len=1) {
(string) (len=12) "preferential": (float64) 1
},
MaxChannels: (int) 5,
Allocation: (float64) 0.6,
MinChannelSize: (int64) 20000,
MaxChannelSize: (int64) 16777215,
Private: (bool) false,
MinConfs: (int32) 0
})

2019-05-16 20:02:22.144 [INF] RPCS: RPC server listening on 127.0.0.1:10009
2019-05-16 20:02:22.144 [INF] RPCS: gRPC proxy started at 127.0.0.1:8080
2019-05-16 20:02:22.146 [INF] LTND: Waiting for chain backend to finish sync, start_height=576376
2019-05-16 20:02:22.146 [INF] RPCS: Stopping RPC Server
2019-05-16 20:02:22.146 [INF] RPCS: Stopping SignRPC Sub-RPC Server
2019-05-16 20:02:22.146 [INF] RPCS: Stopping ChainRPC Sub-RPC Server
2019-05-16 20:02:22.146 [INF] RPCS: Stopping InvoicesRPC Sub-RPC Server
2019-05-16 20:02:22.146 [INF] RPCS: Stopping WalletKitRPC Sub-RPC Server
2019-05-16 20:02:22.146 [INF] LTND: Shutdown complete
2019-05-16 20:02:25.104 [INF] LTND: Version: 0.6.0-beta commit=v0.6-beta, build=production, logging=default
2019-05-16 20:02:25.104 [INF] LTND: Active chain: Bitcoin (network=mainnet)
2019-05-16 20:02:25.105 [INF] CHDB: Checking for schema update: latest_version=8, db_version=8
2019-05-16 20:02:25.118 [INF] RPCS: password RPC server listening on 127.0.0.1:10009
2019-05-16 20:02:25.118 [INF] RPCS: password gRPC proxy started at 127.0.0.1:8080
2019-05-16 20:02:25.118 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create a wallet, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.
2019-05-16 20:02:44.771 [INF] LNWL: Opened wallet
2019-05-16 20:02:44.893 [INF] LTND: Primary chain is set to: bitcoin
2019-05-16 20:02:44.925 [INF] LTND: Initializing bitcoind backed fee estimator
2019-05-16 20:02:44.925 [INF] LNWL: Started listening for bitcoind transaction notifications via ZMQ on tcp://127.0.0.1:18501
2019-05-16 20:02:44.925 [INF] LNWL: Started listening for bitcoind block notifications via ZMQ on tcp://127.0.0.1:18500
2019-05-16 20:02:45.728 [INF] LNWL: The wallet has been unlocked without a time limit
2019-05-16 20:02:45.731 [INF] LTND: LightningWallet opened
2019-05-16 20:02:45.790 [INF] LNWL: Started rescan from block 00000000000000000013e3ddf285abd052fca080c09e79bef1d637405d13ec37 (height 564672) for 24 addresses
2019-05-16 20:02:45.790 [INF] LNWL: Starting rescan from block 00000000000000000013e3ddf285abd052fca080c09e79bef1d637405d13ec37
2019-05-16 20:02:45.793 [INF] HSWC: Restoring in-memory circuit state from disk
2019-05-16 20:02:45.798 [INF] LNWL: Rescan finished at 564672 (00000000000000000013e3ddf285abd052fca080c09e79bef1d637405d13ec37)
2019-05-16 20:02:45.798 [ERR] LNWL: Unable to complete chain rescan: -1: Block not available (pruned data)
2019-05-16 20:02:45.798 [INF] LNWL: Catching up block hashes to height 564672, this might take a while
2019-05-16 20:02:45.813 [INF] LTND: Received terminated
2019-05-16 20:02:45.817 [INF] LTND: Shutting down...
2019-05-16 20:02:45.816 [INF] HSWC: Payment circuits loaded: num_pending=0, num_open=0
2019-05-16 20:02:45.817 [INF] LTND: Gracefully shutting down.
2019-05-16 20:02:45.819 [INF] HSWC: Trimming open circuits for chan_id=562360:3077:0, start_htlc_id=0
2019-05-16 20:02:45.838 [INF] HSWC: Trimming open circuits for chan_id=562236:1668:0, start_htlc_id=26
2019-05-16 20:02:45.837 [INF] LNWL: Done catching up block hashes
2019-05-16 20:02:45.839 [INF] LNWL: Finished rescan for 24 addresses (synced to block 00000000000000000013e3ddf285abd052fca080c09e79bef1d637405d13ec37, height 564672)
2019-05-16 20:02:45.946 [INF] LTND: Channel backup proxy channel notifier starting
2019-05-16 20:02:45.946 [INF] ATPL: Instantiating autopilot with cfg: (*main.autoPilotConfig)(0xc000040680)({
Active: (bool) false,
Heuristic: (map[string]float64) (len=1) {
(string) (len=12) "preferential": (float64) 1
},
MaxChannels: (int) 5,
Allocation: (float64) 0.6,
MinChannelSize: (int64) 20000,
MaxChannelSize: (int64) 16777215,
Private: (bool) false,
MinConfs: (int32) 0
})

2019-05-16 20:02:45.947 [INF] RPCS: RPC server listening on 127.0.0.1:10009
2019-05-16 20:02:45.947 [INF] RPCS: gRPC proxy started at 127.0.0.1:8080
2019-05-16 20:02:45.953 [INF] LTND: Waiting for chain backend to finish sync, start_height=576376
2019-05-16 20:02:45.953 [INF] RPCS: Stopping RPC Server
2019-05-16 20:02:45.953 [INF] RPCS: Stopping InvoicesRPC Sub-RPC Server
2019-05-16 20:02:45.953 [INF] RPCS: Stopping WalletKitRPC Sub-RPC Server
2019-05-16 20:02:45.953 [INF] RPCS: Stopping SignRPC Sub-RPC Server
2019-05-16 20:02:45.953 [INF] RPCS: Stopping ChainRPC Sub-RPC Server
2019-05-16 20:02:45.953 [INF] LTND: Shutdown complete
2019-05-16 20:02:49.009 [INF] LTND: Version: 0.6.0-beta commit=v0.6-beta, build=production, logging=default
2019-05-16 20:02:49.009 [INF] LTND: Active chain: Bitcoin (network=mainnet)
2019-05-16 20:02:49.010 [INF] CHDB: Checking for schema update: latest_version=8, db_version=8
2019-05-16 20:02:49.020 [INF] RPCS: password RPC server listening on 127.0.0.1:10009
2019-05-16 20:02:49.020 [INF] RPCS: password gRPC proxy started at 127.0.0.1:8080
2019-05-16 20:02:49.020 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create a wallet, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.
2019-05-16 20:03:12.083 [INF] LNWL: Opened wallet
2019-05-16 20:03:12.200 [INF] LTND: Primary chain is set to: bitcoin
2019-05-16 20:03:12.234 [INF] LTND: Initializing bitcoind backed fee estimator
2019-05-16 20:03:12.234 [INF] LNWL: Started listening for bitcoind transaction notifications via ZMQ on tcp://127.0.0.1:18501
2019-05-16 20:03:12.234 [INF] LNWL: Started listening for bitcoind block notifications via ZMQ on tcp://127.0.0.1:18500
2019-05-16 20:03:13.091 [INF] LNWL: The wallet has been unlocked without a time limit
2019-05-16 20:03:13.094 [INF] LTND: LightningWallet opened
2019-05-16 20:03:13.156 [INF] HSWC: Restoring in-memory circuit state from disk
2019-05-16 20:03:13.175 [INF] LNWL: Started rescan from block 00000000000000000013e3ddf285abd052fca080c09e79bef1d637405d13ec37 (height 564672) for 24 addresses
2019-05-16 20:03:13.175 [INF] LNWL: Starting rescan from block 00000000000000000013e3ddf285abd052fca080c09e79bef1d637405d13ec37
2019-05-16 20:03:13.181 [INF] HSWC: Payment circuits loaded: num_pending=0, num_open=0
2019-05-16 20:03:13.184 [INF] HSWC: Trimming open circuits for chan_id=562360:3077:0, start_htlc_id=0
2019-05-16 20:03:13.184 [INF] HSWC: Trimming open circuits for chan_id=562236:1668:0, start_htlc_id=26
2019-05-16 20:03:13.186 [INF] LNWL: Rescan finished at 564672 (00000000000000000013e3ddf285abd052fca080c09e79bef1d637405d13ec37)
2019-05-16 20:03:13.186 [ERR] LNWL: Unable to complete chain rescan: -1: Block not available (pruned data)
2019-05-16 20:03:13.186 [INF] LNWL: Catching up block hashes to height 564672, this might take a while
2019-05-16 20:03:13.207 [INF] LTND: Received terminated
2019-05-16 20:03:13.208 [INF] LTND: Shutting down...
2019-05-16 20:03:13.208 [INF] LTND: Gracefully shutting down.
2019-05-16 20:03:13.254 [INF] LNWL: Done catching up block hashes
2019-05-16 20:03:13.254 [INF] LNWL: Finished rescan for 24 addresses (synced to block 00000000000000000013e3ddf285abd052fca080c09e79bef1d637405d13ec37, height 564672)
2019-05-16 20:03:13.313 [INF] LTND: Channel backup proxy channel notifier starting
2019-05-16 20:03:13.313 [INF] ATPL: Instantiating autopilot with cfg: (*main.autoPilotConfig)(0xc000040740)({
Active: (bool) false,
Heuristic: (map[string]float64) (len=1) {
(string) (len=12) "preferential": (float64) 1
},
MaxChannels: (int) 5,
Allocation: (float64) 0.6,
MinChannelSize: (int64) 20000,
MaxChannelSize: (int64) 16777215,
Private: (bool) false,
MinConfs: (int32) 0
})

2019-05-16 20:03:13.315 [INF] RPCS: RPC server listening on 127.0.0.1:10009
2019-05-16 20:03:13.316 [INF] RPCS: gRPC proxy started at 127.0.0.1:8080
2019-05-16 20:03:13.318 [INF] LTND: Waiting for chain backend to finish sync, start_height=576376
2019-05-16 20:03:13.318 [INF] RPCS: Stopping RPC Server
2019-05-16 20:03:13.318 [INF] RPCS: Stopping SignRPC Sub-RPC Server
2019-05-16 20:03:13.318 [INF] RPCS: Stopping ChainRPC Sub-RPC Server
2019-05-16 20:03:13.318 [INF] RPCS: Stopping InvoicesRPC Sub-RPC Server
2019-05-16 20:03:13.318 [INF] RPCS: Stopping WalletKitRPC Sub-RPC Server
2019-05-16 20:03:13.318 [INF] LTND: Shutdown complete
2019-05-16 20:03:16.508 [INF] LTND: Version: 0.6.0-beta commit=v0.6-beta, build=production, logging=default
2019-05-16 20:03:16.508 [INF] LTND: Active chain: Bitcoin (network=mainnet)
2019-05-16 20:03:16.508 [INF] CHDB: Checking for schema update: latest_version=8, db_version=8
2019-05-16 20:03:16.518 [INF] RPCS: password RPC server listening on 127.0.0.1:10009
2019-05-16 20:03:16.518 [INF] RPCS: password gRPC proxy started at 127.0.0.1:8080
2019-05-16 20:03:16.518 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create a wallet, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.`

**Comment by corbimos** [comment]
Ok so i think i figured it out. 

It was shutting down because of me only having a pruned node. It not having all the blocks archived made it fail scanning the wallet addressed... I think?

`2019-05-16 20:03:13.186 [ERR] LNWL: Unable to complete chain rescan: -1: Block not available (pruned data)
2019-05-16 20:03:13.186 [INF] LNWL: Catching up block hashes to height 564672, this might take a while
2019-05-16 20:03:13.207 [INF] LTND: Received terminated
2019-05-16 20:03:13.208 [INF] LTND: Shutting down...
2 `

I have a full archival node running at home, so I copied the entire bitcoin folder from that machine to a hard drive and then pointed my bitcoin directory of the node launcher machine to the external drive's bitcoin directory. 

At that point I was able to unlock the wallet through the command line 

`"/Users/*USER*/Library/Application Support/Node Launcher/bin/lncli" unlock` 

It took a bit for lnd to sync and i saw some errors like the one below, but i see that is normal while lnd is syncing the chain.
`[ERR] LNWL: Failed to process consensus server notification`

The wallet synced just fine and i was able to get to my coins. However, it does not play well if i switch back to a pruned node. I just moved my coins out and trashed this wallet. 

The node launcher seems to run fine with a new wallet on a pruned node. I don't know why my old wallet stopped working.
